### DISCUSSION TITLE
Hide features on target node

### DISCUSSION THREAD

**Discussion by jakewu** [pyg-team/pytorch_geometric#4475] [Q&A]
I’m working on a node classification problem to predict which users are likely to make a purchase and become a customer. The input includes a social graph with edges indicating user-to-user friendship, and node features such as demographic and purchase history of all users. The idea of using graph learning here is that for a user with no purchase history, if most of her friends have made purchases, she should be more likely to convert to a customer in the future. 

My thinking is during training, we should hide purchase history features for the target node itself. Otherwise, the model will trivially learn to predict customer based off the existence of purchase history. On the other hand, we can use the full feature set including purchase history from their friends per the idea mentioned above. 

If all of these make sense, my question is how to hide certain features for the target node during training? Do I need to use [heterogeneous graph learning](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html) and treat the target node and neighboring node as two different types? Any code sample, papers, or discussion pointers would be appreciated!

**Comment by rusty1s** [comment]
Interesting. I don‘t think you need to represent them as different types. It should be sufficient to just temporary remove the links to the purchase history of nodes you are currently training from. I think this is similar to the `examples/unimp_arxiv.py` example which masks out labels for certain training nodes (in your case, you would mask out purchases).
